src/jit/codegenxarch.cpp

```diff
@@ -188,7 +224,13 @@ void                CodeGen::genEmitGSCookieCheck(bool pushReg)
         if (compiler->lvaKeepAliveAndReportThis() && compiler->lvaTable[compiler->info.compThisArg].lvRegister &&
             (compiler->lvaTable[compiler->info.compThisArg].lvRegNum == REG_ECX))
         {
```

Comment:
Ok. Lubo offline pointed out that there could be a bug if thisptr is moved from edi to ecx in method prolog and kept alive throughout the method and such a method is returning a struct containing gc-refs in two return registers.